pennylane_lightning/core/src/simulators/lightning_kokkos/measurements/MeasuresFunctors.hpp

```diff
@@ -16,27 +16,426 @@
+                const std::size_t rev_wire_3, const std::size_t rev_wire_4,
+                const std::size_t rev_wire_5, const std::size_t rev_wire_6,
+                const std::size_t offset, PrecisionT dst[]) const {
+        apply5(i0, rev_wire_0, rev_wire_1, rev_wire_2, rev_wire_3, rev_wire_4,
+               rev_wire_5, 0 + offset, dst);
+        apply5(i0 | (1U << rev_wire_6), rev_wire_0, rev_wire_1, rev_wire_2,
+               rev_wire_3, rev_wire_4, rev_wire_5, 64 + offset, dst);
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    void apply7(const std::size_t i0, const std::size_t rev_wire_0,
+                const std::size_t rev_wire_1, const std::size_t rev_wire_2,
+                const std::size_t rev_wire_3, const std::size_t rev_wire_4,
+                const std::size_t rev_wire_5, const std::size_t rev_wire_6,
+                const std::size_t rev_wire_7, const std::size_t offset,
+                PrecisionT dst[]) const {
+        apply6(i0, rev_wire_0, rev_wire_1, rev_wire_2, rev_wire_3, rev_wire_4,
+               rev_wire_5, rev_wire_6, 0 + offset, dst);
+        apply6(i0 | (1U << rev_wire_7), rev_wire_0, rev_wire_1, rev_wire_2,
+               rev_wire_3, rev_wire_4, rev_wire_5, rev_wire_6, 128 + offset,
+               dst);
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    void apply8(const std::size_t i0, const std::size_t rev_wire_0,
+                const std::size_t rev_wire_1, const std::size_t rev_wire_2,
+                const std::size_t rev_wire_3, const std::size_t rev_wire_4,
+                const std::size_t rev_wire_5, const std::size_t rev_wire_6,
+                const std::size_t rev_wire_7, const std::size_t rev_wire_8,
+                const std::size_t offset, PrecisionT dst[]) const {
+        apply7(i0, rev_wire_0, rev_wire_1, rev_wire_2, rev_wire_3, rev_wire_4,
+               rev_wire_5, rev_wire_6, rev_wire_7, 0 + offset, dst);
+        apply7(i0 | (1U << rev_wire_8), rev_wire_0, rev_wire_1, rev_wire_2,
+               rev_wire_3, rev_wire_4, rev_wire_5, rev_wire_6, rev_wire_7,
+               256 + offset, dst);
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    void operator()(std::size_t k, PrecisionT dst[]) const {
+        if constexpr (num_wires == 0) {
+            std::size_t i0 = (k & parity[0]);
+            for (std::size_t i = 1; i < n_wires + 1; i++) {
+                i0 |= ((k << i) & parity[i]);
+            }
+            for (std::size_t inner_idx = 0; inner_idx < value_count;
+                 inner_idx++) {
+                std::size_t idx = i0;
+                for (std::size_t i = 0; i < n_wires; i++) {
+                    idx |= ((inner_idx & (one << i)) >> i) << rev_wires[i];
+                }
+                const PrecisionT rsv = real(arr(idx));
+                const PrecisionT isv = imag(arr(idx));
+                dst[inner_idx] += rsv * rsv + isv * isv;
+            }
+        }
+        if constexpr (num_wires == 1) {
```

Comment:
Specialized cases below initialize `i0` and call `applyX` of the appropriate recursion level.